keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/ec2/resource/SecurityGroupHandler.kt

```diff
@@ -333,7 +334,7 @@ class SecurityGroupHandler(
           // we have to do a 2-phase create for self-referencing ingress rules as the referenced
           // security group must exist prior to the rule being applied. We filter then out here and
           // the subsequent diff will apply the additional group(s).
-          .filterNot { it.isSelfReference }
+          .filterNot { it is ReferenceRule && it.name == moniker.name }
```

Comment:
Because I had to remove that property from `SecurityGroupRule`

Reply:
I'm struggling a bit to understand how this would address the case we saw with Adam's delivery config, where the name he used did not seem to be derived from the moniker (name was just `spinnakerstatus`). Or was it just that he had no stack in the moniker?

Reply:
correct